device/cmos.go

```diff
@@ -0,0 +1,127 @@
+package device
+
+import (
+	"time"
+)
+
+const (
+	indexMask   = uint8(0x7F)
+	indexOffset = uint64(0x70)
+	dataOffset  = uint64(0x71)
+	dataLen     = uint64(128)
+)
+
+type CMOS struct {
+	Index uint8
+	Data  []uint8
+}
+
+func NewCMOS(memBelow4G, memAbove4G uint64) *CMOS {
+	cmos := &CMOS{
+		Index: 0,
+		Data:  make([]uint8, dataLen),
+	}
+
+	// We assume 3G RAM at all times for now.....
+	extMem := uint16(0xBC00)
+
+	cmos.Data[0x34] = uint8(extMem)
+	cmos.Data[0x35] = uint8(extMem >> 8)
+
+	// Only valid for PVH boot of firmware with 3G RAM fixed.....
+	cmos.Data[0x5b] = 0
+	cmos.Data[0x5c] = 0
+	cmos.Data[0x5d] = 0
+
+	return cmos
+}
+
+func (c *CMOS) Read(base uint64, data []byte) error {
+	if len(data) != 1 {
+		return errDataLenInvalid
+	}
+
+	var d uint8
+
+	// Reading CMOS RAM also requires two steps:
+	// 1. OUT to port hex 70 with the CMOS address that is to be read from.
+	// 2. IN from port hex 71, and the data read is returned in the AL register.
+	// Ref: http://bitsavers.trailing-edge.com/pdf/ibm/pc/at/1502494_PC_AT_Technical_Reference_Mar84.pdf
```

Comment:
Thanks for clear explaination :)

Reply:
Sure thing. The huge pain with kvm im general is the fact that documentation is scattered through the internet.

If we can get some stuff together with better comments on code, I think someone else will highly appreciate it :)

Reply:
Yeah, I think so too.